crates/sargon/src/signing/petition_types/petition_for_factors_types/petition_for_factors/petition_for_factors_sub_state.rs

```diff
@@ -2,38 +2,106 @@ use crate::prelude::*;
 
 /// A sub-state of `PetitionForFactorsState` which can be used to track factors
 /// that have signed or skipped.
-#[derive(Clone, PartialEq, Eq, derive_more::Debug)]
-#[debug("[{}]", factors.borrow().clone().into_iter().map(|f| format!("{:?}", f)).join(", "))]
+#[derive(derive_more::Debug)]
+#[debug("[{}]", self.snapshot().into_iter().map(|f| format!("{:?}", f)).join(", "))]
 pub(crate) struct PetitionForFactorsSubState<F>
 where
-    F: FactorSourceReferencing + Debug,
+    F: FactorSourceReferencing + Debug + HasSampleValues,
```

Comment:
try to see if you can drop `+ HasSampleValues` here, I think you should. It should only be needed below:

```rust
impl<F: FactorSourceReferencing + Debug + HasSampleValues> HasSampleValues
    for PetitionForFactorsSubState<F>
{
}
```

And if you need it elsewhere, in macros in `sargon-uniffi` crate, try putting the impl in a `impl<T> HasSampleValues for Struct<X> where X: HasSampleValues`.